src/expandableCalendar/index.tsx

```diff
@@ -91,515 +96,491 @@ interface State {
 
     return (
       <Animated.View
-        ref={this.weekCalendar}
-        style={[this.style.weekContainer, position === Positions.OPEN ? this.style.hidden : this.style.visible]}
+        ref={weekCalendar}
+        style={[style.current.weekContainer, isOpen ? style.current.hidden : style.current.visible]}
         pointerEvents={position === Positions.CLOSED ? 'auto' : 'none'}
       >
         <WeekComponent
-          {...this.props}
+          {...props}
           {...weekCalendarProps}
-          current={this.props.context.date}
-          onDayPress={this.onDayPress}
-          style={this.props.calendarStyle}
+          current={date}
+          onDayPress={_onDayPress}
+          style={calendarStyle}
           hideDayNames={true}
           accessibilityElementsHidden // iOS
           importantForAccessibility={'no-hide-descendants'} // Android
         />
       </Animated.View>
     );
-  }
+  };
 
-  renderKnob() {
+  const renderKnob = () => {
     // TODO: turn to TouchableOpacity with onPress that closes it
     return (
-      <View style={this.style.knobContainer} pointerEvents={'none'} testID={`${this.props.testID}-knob`}>
-        <View style={this.style.knob} testID={CALENDAR_KNOB} />
+      <View style={style.current.knobContainer} pointerEvents={'none'} testID={`${testID}-knob`}>
+        <View style={style.current.knob} testID={CALENDAR_KNOB}/>
       </View>
     );
-  }
-
-  renderArrow = (direction: Direction) => {
-    const {renderArrow, rightArrowImageSource = RIGHT_ARROW, leftArrowImageSource = LEFT_ARROW, testID} = this.props;
+  };
 
+  const _renderArrow = (direction: Direction) => {
     if (isFunction(renderArrow)) {
       return renderArrow(direction);
     }
 
     return (
       <Image
         source={direction === 'right' ? rightArrowImageSource : leftArrowImageSource}
-        style={this.style.arrowImage}
+        style={style.current.arrowImage}
         testID={`${testID}-${direction}-arrow`}
       />
     );
   };
 
-  render() {
-    const {style, hideKnob, horizontal, allowShadow, theme, ...others} = this.props;
-    const {deltaY, position, screenReaderEnabled} = this.state;
-    const isOpen = position === Positions.OPEN;
-    const themeObject = Object.assign(this.headerStyleOverride, theme);
-
-    return (
-      <View testID={this.props.testID} style={[allowShadow && this.style.containerShadow, style]}>
-        {screenReaderEnabled ? (
-          <Calendar
+  return (
+    <View testID={testID} style={[allowShadow && style.current.containerShadow, propsStyle]}>
+      {screenReaderEnabled ? (
+        <Calendar
+          testID="calendar"
+          {...others}
+          theme={themeObject}
+          onDayPress={_onDayPress}
+          hideExtraDays
+          renderArrow={_renderArrow}
+        />
+      ) : (
+        <Animated.View ref={wrapper} style={{height: deltaY.current}} {...panResponder.panHandlers}>
```

Comment:
consider memoizing even that it depends on a ref object. 
The object here always recreated